src/Symfony/Component/HttpKernel/EventListener/CacheAttributeListener.php

```diff
@@ -0,0 +1,194 @@
+        }
+        $response = $event->getResponse();
+
+        // http://tools.ietf.org/html/draft-ietf-httpbis-p4-conditional-12#section-3.1
+        if (!\in_array($response->getStatusCode(), [200, 203, 300, 301, 302, 304, 404, 410])) {
+            unset($this->lastModified[$request]);
+            unset($this->etags[$request]);
+
+            return;
+        }
+
+        if (isset($this->lastModified[$request]) && !$response->headers->has('Last-Modified')) {
+            $response->setLastModified($this->lastModified[$request]);
+        }
+
+        if (isset($this->etags[$request]) && !$response->headers->has('Etag')) {
+            $response->setEtag($this->etags[$request]);
+        }
+
+        unset($this->lastModified[$request]);
+        unset($this->etags[$request]);
+        $hasVary = $response->headers->has('Vary');
+
+        foreach (array_reverse($attributes) as $cache) {
+            if (null !== $cache->smaxage && !$response->headers->hasCacheControlDirective('s-maxage')) {
+                $response->setSharedMaxAge($this->toSeconds($cache->smaxage));
+            }
+
+            if ($cache->mustRevalidate) {
+                $response->headers->addCacheControlDirective('must-revalidate');
+            }
+
+            if (null !== $cache->maxage && !$response->headers->hasCacheControlDirective('max-age')) {
+                $response->setMaxAge($this->toSeconds($cache->maxage));
+            }
+
+            if (null !== $cache->maxStale && !$response->headers->hasCacheControlDirective('max-stale')) {
+                $response->headers->addCacheControlDirective('max-stale', $this->toSeconds($cache->maxStale));
+            }
+
+            if (null !== $cache->staleWhileRevalidate && !$response->headers->hasCacheControlDirective('stale-while-revalidate')) {
+                $response->headers->addCacheControlDirective('stale-while-revalidate', $this->toSeconds($cache->staleWhileRevalidate));
+            }
+
+            if (null !== $cache->staleIfError && !$response->headers->hasCacheControlDirective('stale-if-error')) {
+                $response->headers->addCacheControlDirective('stale-if-error', $this->toSeconds($cache->staleIfError));
+            }
+
+            if (null !== $cache->expires && !$response->headers->has('Expires')) {
+                $response->setExpires(new \DateTimeImmutable('@'.strtotime($cache->expires, time())));
+            }
+
+            if (!$hasVary && $cache->vary) {
+                $response->setVary($cache->vary, false);
+            }
+        }
+
+        foreach ($attributes as $cache) {
+            if (true === $cache->public) {
+                $response->setPublic();
+            }
+
+            if (false === $cache->public) {
+                $response->setPrivate();
+            }
+        }
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::CONTROLLER => ['onKernelController', 10],
```

Comment:
Could we start using meaningful method name in listeners? IIRC this naming convention was to automate binding methods to events.

The name of this event already gives the "when" context, the method should give the "what" context.
```suggestion
            KernelEvents::CONTROLLER => ['readCacheConfiguration', 10],
            KernelEvents::RESPONSE => ['setCache', -10],
```

Reply:
I'd prefer not, this still takes an event as argument.

Reply:
Sorry, I really miss your point.

Allow me give more context to better compare the two:
```diff
class CacheAttributeListener implements EventSubscriberInterface

-    public function onKernelController(ControllerEvent $event)
+    public function readCacheConfiguration(ControllerEvent $event)

-    public function onKernelResponse(ResponseEvent $event)
+    public function setCache(ResponseEvent $event)

    public static function getSubscribedEvents(): array
-        KernelEvents::CONTROLLER => ['onKernelController', 10]
+        KernelEvents::CONTROLLER => ['readCacheConfiguration', 10]
-        KernelEvents::RESPONSE => ['onKernelResponse', -10]
+        KernelEvents::RESPONSE => ['setCache', -10]
```

The current form looks like a [tautology](https://en.wikipedia.org/wiki/Tautology_(language)) without useful meaning.

Reply:
I agree with @HeahDude

Reply:
I'm sorry I don't. Because it's an event listener, it's more important to me to clearly state when a method is called than what it does. The *what* is conveyed by the name of the class to me. I don't think that we made a mistake in the past by naming methods by their wiring position.

Reply:
The "when" is already conveyed by both the configuration and the argument.

In case of debugging, it clearly brings no value, especially when the same class is used with multiple events (like this one), or multiple priorities (like the `LocaleListener`):

<img width="817" alt="Screenshot 2022-07-11 at 9 35 27 AM" src="https://user-images.githubusercontent.com/10107633/178212626-0fde19b4-ee89-4898-8725-122afbad88d5.png">

Reply:
A good example is `TraceableFirewallListener::configureLogoutUrlGenerator()`: from a public API pov, this method name makes little sense to me. If one day we decide to make this class do more on kernel.request, we're going to have to deal with a deprecation layer just to keep the name aligned with what does? Better not.

Reply:
`TraceableFirewallListener` is a good exemple where two methods doing two different things need two different names.

>If one day we decide to make this class do more on kernel.request, we're going to have to deal with a deprecation layer just to keep the name aligned with what it does?

If a method do something else to the point that its name is not good anymore maybe it needs a new dedicated one? It's theoretical, unless you have a concrete exemple in mind.

I'd like to mention that the opposite is not less complex, when the listened event is changed (could even be as bug fix), whereas it does the same thing.
I've met a similar case a long time ago https://github.com/symfony/symfony/pull/18069.

Reply:
We could also decide to make listener callables final to not be constrained either way.